compiler-plugin/src/main/java/io/ballerina/stdlib/kafka/plugin/KafkaFunctionValidator.java

```diff
@@ -93,26 +97,54 @@ private void validateOnConsumerRecord() {
     private void validateFunctionParameters(SeparatedNodeList<ParameterNode> parameters,
                                             FunctionDefinitionNode functionDefinitionNode) {
         if (parameters.size() == 1) {
```

Comment:
Shall we put a comment on the compiler-validation logic? It is bit hard to understand and remember the context when we read it top-to-bottom.

Reply:
Added in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/370/commits/cc7b1c400a321a7bb8cb6358485eb50fff38c9b0